### DISCUSSION TITLE
Hello, can someone help me with the type problem

### DISCUSSION THREAD

**Discussion by RustLoverTheCoder** [pmndrs/jotai#2527] [Q&A] [answered]
![image](https://github.com/pmndrs/jotai/assets/71162666/edebc09a-7fd7-46e4-b724-e8c254e63e14)

**Comment by dai-shi** [comment]
If it's a type issue, a small reproduction with https://www.typescriptlang.org/play might be useful for someone to help.
It looks like `settingAtom` is async, so that should be the issue. So, it's not only a type issue.

**Reply by RustLoverTheCoder** [reply]
Yes, settingAtom is an asynchronous atom. How do I solve this problem

**Reply by dai-shi** [reply, marked as answer]
`unwrap` util might help.
Or, you could make your reducer async, but it might not be very common.

**Reply by RustLoverTheCoder** [reply]
Thank you very much, I really like jotai's various atomic combinations
